Sources/web3swift/Utils/ENS/ENSResolver.swift

```diff
@@ -62,28 +62,43 @@ public extension ENS {
         }
 
         public func supportsInterface(interfaceID: Data) async throws -> Bool {
-            guard let supports = try? await supportsInterface(interfaceID: interfaceID.toHexString()) else {throw Web3Error.processingError(desc: "Can't get answer")}
+            guard let supports = try? await supportsInterface(interfaceID: interfaceID.toHexString()) else {
+                throw Web3Error.processingError(desc: "Can't get answer")
+            }
+            return supports
+        }
+
+        public func supportsInterface(interfaceID: InterfaceName) async throws -> Bool {
+            guard let supports = try? await supportsInterface(interfaceID: interfaceID.hash) else {
+                throw Web3Error.processingError(desc: "Can't get answer")
+            }
             return supports
         }
 
         public func supportsInterface(interfaceID: String) async throws -> Bool {
-            guard let transaction = self.resolverContract.createReadOperation("supportsInterface", parameters: [interfaceID as AnyObject], extraData: Data() ) else {throw Web3Error.transactionSerializationError}
-            guard let result = try? await transaction.callContractMethod() else {throw Web3Error.processingError(desc: "Can't call transaction")}
-            guard let supports = result["0"] as? Bool else {throw Web3Error.processingError(desc: "Can't get answer")}
+            guard let transaction = self.resolverContract.createReadOperation("supportsInterface", parameters: [interfaceID as AnyObject]) else {
+                throw Web3Error.transactionSerializationError
+            }
+            guard let result = try? await transaction.callContractMethod() else {
+                throw Web3Error.processingError(desc: "Can't call transaction")
+            }
+            guard let supports = result["0"] as? Bool else {
+                throw Web3Error.processingError(desc: "Can't get answer")
+            }
             return supports
         }
 
         public func interfaceImplementer(forNode node: String, interfaceID: String) async throws -> EthereumAddress {
             guard let nameHash = NameHash.nameHash(node) else {throw Web3Error.processingError(desc: "Failed to get name hash")}
-            guard let transaction = self.resolverContract.createReadOperation("interfaceImplementer", parameters: [nameHash, interfaceID] as [AnyObject], extraData: Data() ) else {throw Web3Error.transactionSerializationError}
+            guard let transaction = self.resolverContract.createReadOperation("interfaceImplementer", parameters: [nameHash, interfaceID] as [AnyObject]) else {throw Web3Error.transactionSerializationError}
```

Comment:
@janndriessen @yaroslavyaroslav I've decided to remove from all `createReadOperation` and `createWriteOperation` calls the `extraData: Data()` arguments as `Data()` is the default value for the `extraData` in these functions.

Reply:
nice and clean 🧹 ✅